core/tx_pool.go

```diff
@@ -1205,6 +1222,12 @@ func (pool *TxPool) reset(oldHead, newHead *types.Header) {
 	next := new(big.Int).Add(newHead.Number, big.NewInt(1))
 	pool.istanbul = pool.chainconfig.IsIstanbul(next)
 	pool.eip2718 = pool.chainconfig.IsBerlin(next)
+	pool.eip1559 = pool.chainconfig.IsAleut(next)
+
+	// For EIP-1559, adjust the gas limit by the elasticity multiplier
+	if pool.eip1559 {
+		pool.currentMaxGas *= params.ElasticityMultiplier
```

Comment:
I think it's easier to set this explicitly 
```suggestion
		pool.currentMaxGas = newHead.GasLimit * params.ElasticityMultiplier
```